tensorflow-tools/src/main/java/org/tensorflow/tools/Shape.java

```diff
@@ -61,21 +74,57 @@ public static Shape of(long... dimensionSizes) {
     return new Shape(dimensionSizes);
   }
 
+  /**
+   * Returns the total number of elements a Tensor with this shape would have. If this
+   * {@link Shape#isUnknown()} is true or {@link Shape#hasUnknownDimension()}  is true,
+   * {@link Shape#UNKNOWN_SIZE} is returned.
+   *
+   * @return the total number of elements a Tensor with this shape would have if it can be
+   * calculated, else {@link Shape#UNKNOWN_SIZE}.
+   */
   public long size() {
     if (size == null) {
       size = computeSize(dimensionSizes);
     }
     return size;
   }
 
+  /**
+   * The size of the dimension with the given index. If {@link Shape#isUnknown()} is true or
+   * the size of the dimension with the given index has an unknown size, {@link Shape#UNKNOWN_SIZE}
+   * is returned.
+   * @param i the index of the dimension to get the size for. If this Shape has a known number of
+   *          dimensions, it must be < {@link Shape#numDimensions()}. The index may be negative,
+   *          in which case the position is counted from the end of the shape. E.g.
+   *          {@code size(-1) returns the size of the last dimension, size(-2) the size of the
+   *          second to last dimension etc.}
+   * @return The size of the dimension with the given index if known, {@link Shape#UNKNOWN_SIZE}
+   *         otherwise.
+   */
   public long size(int i) {
-    return dimensionSizes != null ? dimensionSizes[i] : UNKNOWN_SIZE;
+    if (dimensionSizes == null) {
+      return UNKNOWN_SIZE;
+    } else if (i >= 0) {
+      return dimensionSizes[i];
+    } else {
+      return dimensionSizes[dimensionSizes.length + i];
+    }
   }
 
+  /**
+   * Returns the number of dimensions of this shape. -1 if unknown, 0 for a scalar, 1 for a vector,
+   * 2 for a matrix etc.
+   * @return the number of dimensions of this shape. -1 if unknown, 0 for a scalar, 1 for a vector,
```

Comment:
again according to the [guide](https://google.github.io/styleguide/javaguide.html#s7.2-summary-fragment), you should omit the `@return` tag. Same for other block below.

Reply:
I prefer that anyway, should have removed all of them now.